lib/chef/knife.rb

```diff
@@ -375,7 +375,7 @@ def merge_configs
     #   - :cli_default - came from a declared CLI `option`'s `default` value.
     #   - nil - if they key does not exist
     def config_source(key)
-      return :cli if @original_config.include? key
+      return :cli if config.include? key
```

Comment:
We can resolve this by initializing `@original_config` in the constructor. If `merge_config` is never called, it means `config_source` will report `nil`. 

`config_source` is a new addition to the API; bootstrap is the only one [using it](https://github.com/chef/chef/blob/master/lib/chef/knife/bootstrap.rb#L676-L677), and the effect there will be that we don't show deprecation warnings if deprecated config values are used (as they are, in knife-ec2).